### PR TITLE
update scala to add OMRegisterMap

### PR DESCRIPTION
**Pull request by albertchen-sifive**
regenerate scala with latest version of `duh-scala` and add code to generate `OMRegisterMap`s in user scala. Also update `pio.json5` to match older DUH register schema, because that is what `duh-scala` is currently using.

### CODE REVIEW

File: pio.json5 [hunk 1]
```diff
@@ -103,17 +103,17 @@
           name: 'ODATA',
           addressOffset: 0, size: 32,
           displayName: 'Output Data Register',
-          fields: [{name: 'data', bitOffset: 0, bitWidth: 32}]
```
Comment (richardxia) [inline pio.json5:106]: Just out of curiosity, was this change necessary for correctness, or did you remove it because it was superfluous?

**Review by richardxia — APPROVED** [review]
LGTM!

File: pio.json5 (hunk 1 above)
Comment (albertchen-sifive) [reply pio.json5:106]: necessary since the scala does not generate correctly if you use the wrong field names. `duh-scala` is following an older duh schema.
